apps/ledger-live-desktop/src/renderer/modals/Platform/Exchange/CompleteExchange/Body.tsx

```diff
@@ -142,45 +156,62 @@ const Body = ({ data, onClose }: { data: Data; onClose?: () => void | undefined
 
   const onBroadcastSuccess = useCallback(
     (operation: Operation) => {
-      // If swap we save to swap history and keep open the drawer
-      if (swapId && toAccount && magnitudeAwareRate && sourceCurrency && targetCurrency) {
-        const newResult = {
-          operation,
-          swapId,
-        };
-        updateAccount({
-          result: newResult,
-          magnitudeAwareRate,
-        });
-        setResult({
-          swapId,
-          provider,
-          sourceCurrency,
-          targetCurrency,
-        });
+      // TODO: swapId && sellId will become quoteId
+
+      const isValidExchangeSDKSwapTx =
+        (ExchangeType.SWAP || ExchangeType.SWAP_NG) &&
+        swapId &&
+        toAccount &&
+        magnitudeAwareRate &&
+        sourceCurrency &&
+        targetCurrency;
+
+      const isValidExchangeSDKSellTx =
+        (data.exchangeType === ExchangeType.SELL || data.exchangeType === ExchangeType.SELL_NG) &&
+        sourceCurrency;
+
+      if (isValidExchangeSDKSellTx || isValidExchangeSDKSwapTx) {
+        if (isValidExchangeSDKSwapTx) {
+          const newResult = {
+            operation,
+            swapId,
+          };
+
+          updateAccount({
+            result: newResult,
+            magnitudeAwareRate,
+          });
+        }
+
+        const result = isValidExchangeSDKSwapTx
+          ? { swapId, provider, sourceCurrency, targetCurrency, mode: ExchangeModeEnum.Swap }
+          : { provider, mode: ExchangeModeEnum.Sell, sourceCurrency };
+
+        setResult(result);
```

Comment:
We could send data here, and check if it is "swap" later... what do you think?
```suggestion
        setResult(data);
```